### PR TITLE
Use dnnl for cpu backend

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Review by scxiao — APPROVED** [review]

File: src/include/migraphx/op/contiguous.hpp [hunk 1]
```diff
@@ -44,6 +44,11 @@ struct contiguous
         });
         return result;
     }
+
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply src/include/migraphx/op/contiguous.hpp:47]: Thats a good idea.

[the remark this answers is not in the document]
Comment (pfultz2) [reply src/include/migraphx/op/contiguous.hpp:47]: No we cant get rid of compute as this doesn't inherit from `unary`. There seems to be an issue with using `unary` here. I am not sure what the issue, we can look into this in the future.

**Comment by pfultz2** [comment]
This is passing CI now. @kahmed10 can you review?

**Review by kahmed10 — COMMENTED** [review]
There's an empty file for test/verify/main.cpp? Or is this a github bug?

**Comment by pfultz2** [comment]
> There's an empty file for test/verify/main.cpp? Or is this a github bug?

Its not an empty file(as you can see [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/cpu-target/test/verify/main.cpp)) but it means there are no changes to the file. I am not sure why it shows if there are no changes to the file.

File: src/targets/cpu/include/migraphx/cpu/context.hpp [hunk 2]
```diff
@@ -2,15 +2,65 @@
...
 
+#ifdef USE_DNNL
 struct context
 {
     void finish() const {}
+
+    template <class F>
+    void bulk_execute(std::size_t n, std::size_t min_grain, F f)
+    {
+#if 1
```
Comment (kahmed10) [inline src/targets/cpu/include/migraphx/cpu/context.hpp:21]: Why do we need a macro here?

Comment (pfultz2) [reply src/targets/cpu/include/migraphx/cpu/context.hpp:21]: Yea thats debug code that can be removed.

Comment (pfultz2) [reply src/targets/cpu/include/migraphx/cpu/context.hpp]: Fixed.

**Review by kahmed10 — APPROVED** [review]

**Pull request merged**